### PR TITLE
read all xmp directories

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?
Rather than just `getFirstDirectoryOfType`, read all xmp directories using `getDirectoriesOfType` and iterating over the result.

## How can success be measured?
More metadata extraction.

## Screenshots (if applicable)
n/a

## Who should look at this?

@guardian/digital-cms

## Tested?
- [x] locally
- [ ] on TEST

### CODE REVIEW

**Comment by paperboyo** [comment]
💎 This will allow us to finally migrate our [Getty parser](https://github.com/guardian/grid/blob/master/common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala#L191) to use the presence of `GettyImagesGIFT:AssetId` instead of just the presence of `GettyImagesGIFT` group.

**Comment by ajwl** [comment]
👍 as far as I can reason about it

**Comment by akash1810** [comment]
This is extracting the asset id into the field `prefix0:assetid` rather than `GettyGIFT:assetid`. Needs more work.

**Comment by paperboyo** [comment]
> `prefix0:assetid` rather than `GettyGIFT:assetid`

Boo :-( Why was it OK locally? Is there anything I can do to help? (I can bring tea from our “tea”-point)

**Comment by akash1810** [comment]
Since https://github.com/guardian/grid/pull/2634, this is far more reliable.

File: image-loader/app/lib/imaging/FileMetadataReader.scala [hunk 1]
```diff
@@ -107,24 +110,31 @@ object FileMetadataReader {
...
+
+  private def xmpDirectoryToMap(directory: XmpDirectory, imageId: String): Map[String, String] = {
+    directory.getXmpProperties.asScala.toMap.mapValues(nonEmptyTrimmed).collect {
+      case (datePattern(key), Some(value)) => key -> ImageMetadataConverter.cleanDate(value, key, imageId)
+      case (key, Some(value)) => key -> value
+    }
+  }
+
+  private def exportXmpProperties(metadata: Metadata, imageId:String): Map[String, String] = {
+    val directories = metadata.getDirectoriesOfType(classOf[XmpDirectory]).asScala.toList
```
Comment (akash1810) [inline image-loader/app/lib/imaging/FileMetadataReader.scala]: this is the main change, going from `getFirstDirectoryOfType` to `getDirectoriesOfType`

**Pull request merged**